Mathlib/Logic/Basic.lean

```diff
@@ -429,6 +429,15 @@ theorem rec_heq_iff_heq {α β : Sort _} {a b : α} {C : α → Sort*} {x : C a}
 theorem heq_rec_iff_heq {α β : Sort _} {a b : α} {C : α → Sort*} {x : β} {y : C a} {e : a = b} :
     HEq x (e ▸ y) ↔ HEq x y := by subst e; rfl
 
+universe u
+variable {α β : Sort u} {e : β = α} {a : α} {b : β}
+
+lemma heq_comm : HEq a b ↔ HEq b a := by constructor <;> rintro ⟨⟩ <;> rfl
```

Comment:
This landed in leanprover/lean4#5456 I think

Reply:
Ha! What should I do? It's not on master, and it will very easy to deduplicate the lemma when it is. Should we merge anyway?

Reply:
I guess no action needed, though `-- upstreamed in Lean 4.14` would be reasonable.

Reply:
Sure, sounds reasonable
```suggestion
-- upstreamed in Lean 4.14
lemma heq_comm : HEq a b ↔ HEq b a := by constructor <;> rintro ⟨⟩ <;> rfl
```